stat.c

```diff
@@ -2265,45 +2265,8 @@ static inline void reset_io_stat(struct io_stat *ios)
 void reset_io_stats(struct thread_data *td)
 {
 	struct thread_stat *ts = &td->ts;
-	int i, j;
 
-	for (i = 0; i < DDIR_RWDIR_CNT; i++) {
-		reset_io_stat(&ts->clat_stat[i]);
-		reset_io_stat(&ts->slat_stat[i]);
-		reset_io_stat(&ts->lat_stat[i]);
-		reset_io_stat(&ts->bw_stat[i]);
-		reset_io_stat(&ts->iops_stat[i]);
-
-		ts->io_bytes[i] = 0;
-		ts->runtime[i] = 0;
-		ts->total_io_u[i] = 0;
-		ts->short_io_u[i] = 0;
-		ts->drop_io_u[i] = 0;
-
-		for (j = 0; j < FIO_IO_U_PLAT_NR; j++) {
-			ts->io_u_plat[i][j] = 0;
-			if (!i)
-				ts->io_u_sync_plat[j] = 0;
-		}
-	}
-
-	ts->total_io_u[DDIR_SYNC] = 0;
-
-	for (i = 0; i < FIO_IO_U_MAP_NR; i++) {
-		ts->io_u_map[i] = 0;
-		ts->io_u_submit[i] = 0;
-		ts->io_u_complete[i] = 0;
-	}
-
-	for (i = 0; i < FIO_IO_U_LAT_N_NR; i++)
-		ts->io_u_lat_n[i] = 0;
-	for (i = 0; i < FIO_IO_U_LAT_U_NR; i++)
-		ts->io_u_lat_u[i] = 0;
-	for (i = 0; i < FIO_IO_U_LAT_M_NR; i++)
-		ts->io_u_lat_m[i] = 0;
-
-	ts->total_submit = 0;
-	ts->total_complete = 0;
+	memset(ts, 0, sizeof(*ts));
```

Comment:
It's currently not clearing anything near all of it, however. Might be fine, I'd have to double check, but the commit makes it seems like we're just replacing an all-field-clear with a memset, which would just be a cleanup.

Reply:
I'm not sure whether I need this patch. I will see whether it can be removed.

Reply:
That patch has been removed.